src/tools/terraform.ts

```diff
@@ -0,0 +1,65 @@
+import * as fs from 'fs'
+import * as path from 'path'
+import * as os from 'os'
+import * as unzipper from 'unzipper'
+import {ChildProcess, SpawnOptions} from 'child_process'
+
+// Downloader (maven-helper), install settings and utils
+import {MavenHelper} from './maven-helper'
+import {url, version} from './terraform/settings'
+import {_spawn2} from '../util/child-process'
+import {LoggerFactory} from '../util/logger'
+
+// Tool abstract
+import {Tool} from './tool'
+
+export class Terraform extends Tool {
+  static logger = LoggerFactory.createLogger(Terraform)
+
+  // Get full path to binary ().exe appended for Windows)
+  bin(homeDir: string): string {
+    if (os.platform() === 'win32') return path.join(homeDir, 'terraform.exe')
+    return path.join(homeDir, 'terraform')
+  }
+
+  // Installer
+  async install(): Promise<string> {
+    // Install dir, zip download and bin path
+    const homeDir = path.join(os.tmpdir(), `.nrdk/terraform/${version}`)
+    const zip = path.join(os.tmpdir(), `terraform-${Date.now()}.zip`)
+    const bin = this.bin(homeDir)
+
+    // Return if binary already exists
+    if (fs.existsSync(bin)) return bin
+
+    // Download from url
+    await MavenHelper.downloadFromUrl(new URL(url), zip)
+
+    // Extract
+    const extract = unzipper.Extract
+    return new Promise((resolve, reject) => {
+      fs.createReadStream(zip)
+      .pipe(extract({path: homeDir}))
+      .on('error', reject)
+      .on('finish', () => resolve(true))
```

Comment:
I've run into issues running on Windows. When it runs for the first time, I get
```
    Error: spawn EBUSY
    Code: EBUSY
```
believe that changing from `.on('finish', ...)` to `.on('close', ...)` takes care of that as it seems to be a concurrency problem trying to execute a file that is still being used/locked.